perma_web/frontend/components/CreateLink.vue

```diff
@@ -78,39 +80,68 @@ const handleArchiveRequest = async () => {
             })
 
         if (!response?.ok) {
-            throw new Error(response.statusText) // We will handle this more in-depth later
+            const errorResponse = await getErrorResponse(response)
+            throw errorResponse
         }
 
-        const { guid } = await response.json() // Needed to poll Perma about the capture status of a link
+        const { guid } = await response.json()
         userLinkGUID.value = guid
         globalStore.updateCapture('isQueued')
 
     } catch (error) {
-        globalStore.updateCapture('urlError')
-        globalStore.updateCaptureErrorMessage(error)
+        handleCaptureError({ error, errorType: 'urlError' })
     }
 };
 
+const handleCaptureError = ({ error, errorType }) => {
+    let errorMessage
+
+    // Handle API-generated error messages
+    if (error?.response) {
+        errorMessage = getErrorFromResponseStatus(error.status, error.response)
+    }
+
+    // Handle frontend-generated error messages
+    else if (error.length) {
+        errorMessage = error
+    }
+
+    else if (error?.status) {
+        errorMessage = `Error: ${error.status}`
+    }
+
+    // Handle uncaught errors
+    else {
+        errorMessage = defaultError
+    }
+
+    globalStore.updateCapture(errorType)
+    globalStore.updateCaptureErrorMessage(errorMessage)
+}
+
 const handleCaptureStatus = async (guid) => {
     try {
         const response = await fetch(`/api/v1/user/capture_jobs/${guid}`)
 
         if (!response?.ok) {
-            throw new Error(response.statusText) // We will handle this more in-depth later
+            const errorResponse = await response.json()
```

Comment:
I think this wants to call the new utility `getErrorResponse` too, right, so this code doesn't break if the error is a non-JSON 502, etc. Am I reading that right?

Reply:
Ah, great question! You're totally right — we should be calling `getErrorResponse` here if we want to implement my idea for a `maxFailedAttempts` approach to error handling for this GET request. 

I'll create a Linear issue for a future update to implement that "fancier" approach to handling errors. 

In the interim,  I just pushed a small update in [e960174](https://github.com/harvard-lil/perma/pull/3531/commits/e9601740ed40e8dd501c6564a9a9635d4382f643). The function should now disregard any errors that happen as a result of fetching a job's status, matching the legacy Pema behavior for now.